### ISSUE TITLE
new TPM check

### ISSUE THREAD

**Issue by liwenjie119** [AveYo/MediaCreationTool.bat#57]
In 22509, the new file
C:\$WINDOWS.~BT\Sources\Panther\Appraiser_Data.ini
was checked.

**Comment by AveYo** [comment]
microsoft being microsoft?

**Comment by LillyWho** [comment]
For me it initially got even into the upgrade process, but always failed and rolled back. Now that I started fiddling with these bypass flags, it actually started erroring out explicitly before even initiating the upgrade. The dynamic update script didn't work, but the wim file patch seems to. Currently it's working away at "making sure my PC is ready" whatever that means. It's stressing all hard disks except the actual system drive. Makes sense, not.

**Comment by AveYo** [comment]
v6 standalone script should work more reliably now

**Comment by LillyWho** [comment]
It's still refusing to install for me. Build 22518.1000 came out and it fails at the TPM2.0 check. At least that's what it said? As I type this, I am noticing the download percentage is going up despite the setup window having said it can't install?

**Comment by AveYo** [comment]
@LillyWho, 
are you sure you're running the v6?  just run the script again, it should say:
>Skip TPM Check on Dynamic Update **v6** [INSTALLED] run again to remove

it might get blocked by a zealous antivirus - check your av logs

the whole code is added in the registry here:
`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\SetupHost.exe`
UseFilter = 0x1
`HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\SetupHost.exe\0`
Code= ..
Debugger= ..
FilterFullPath= ..
make sure the entries are there

**Comment by sam20112011** [comment]
The problem appears on reboot and "working with updates". No "work with updates" occurs after clicking restart to install the update. The computer just restarts as usual. It says: Windows 11 Insider Preview 22518.1000 (rs_prerelease) installation error is 0xc1900130 in center update. The update itself is downloaded and completes the installation.

**Comment by sam20112011** [comment]
Entries is fine. No problem or errors with antivirus. Windows 11 mounted iso present itself as Windows Server. I even a deleted a resourses of update and redownload it from Center Update - all works fine until "Work with updates"restart.

**Comment by AveYo** [comment]
Your issue is likely the updating itself. 
Microsoft lists issues installing for some devices and they are trying to address them in upcoming updates.
Mean time you can try the usual windows update troubleshooting, resetting windows update components,  sfc /scannow and Dism /Online /Cleanup-Image /RestoreHealth /Source: ... or running setup from a mounted iso (that you can get for insiders builds via uupdump.net

It would be very useful to understand what's happening, to monitor `C:\$WINDOWS.~BT\Sources` folder
Just before restart you should check that it contains the setup executable files
And should zip the `C:\$WINDOWS.~BT\Sources\Panther` logs folder and share it so I can take a look at

**Comment by sam20112011** [comment]
Hmm... intresting. When i restart updating via Center, during the download and installation process, a lot of files appeared in the C:\$WINDOWS.~BT folder including exe. But upon completion of the installation and the requirement to restart the computer, all that remains in it is this (https://github.com/AveYo/MediaCreationTool.bat/files/7693384/WINDOWS.BT.zip). In addition, the updates Iso created by me also started without problems (he even offered to save all the data on the computer) and did all the necessary checks. But on a blue screen (analogous to "working with updates"), 1% crashed with an error on the desktop "Windows Server installation failed".

**Comment by Core0101** [comment]
Also having the same issue here.
![image](https://user-images.githubusercontent.com/95932132/145612597-6d04f775-6990-431c-be18-d8fe825fad1c.png)
Updates will fail to install. I ran the new V6 script again and tried a second time but got the same result. 
As Windows update is currently sitting at this page my Windows BT folder is empty. SoftwareDistribution has some stuff inside it the folder below was the largest.
![image](https://user-images.githubusercontent.com/95932132/145613017-b3a56cfe-9f57-4e94-b52c-890b5f72a616.png)
I've already tried cleaning out the software distribution folder a couple times and restarting Windows update.

**Comment by AveYo** [comment]
v6**f** update should now finally work for everybody! 
A great help was @sam20112011 providing requested `C:\$WINDOWS.~BT\Sources\Panther` logs

That was showing both `/Product Server` _(bypass installed and active)_ and `/Product Client` _(bypass not working as intended)_, causing two compatibility reports with one obviously failing (the Client one)
Why? because in v6 release I left a bad regex that did not replace launch options correctly. 
Funny thing, the installed bypass in my test machine had the updated regex so it did not fail for me.
Sorry for the frustration caused by this.

**Comment by Core0101** [comment]
Not a problem! Thanks so much for working on this script. My results are below after running the new script. 

So I ran the new script twice (the first one uninstalled the bypass then I ran it again to install it again) and checked for updates without a reboot and it gave the same error. I tried doing a reboot to see if anything changed and it did the same thing. This time I was able to grab the Windows BT folder before it disappeared.

![image](https://user-images.githubusercontent.com/95932132/145644049-1221c746-4f56-4f28-a44f-29652f3aded7.png)

[$WINDOWS.BT.zip](https://github.com/AveYo/MediaCreationTool.bat/files/7695577/WINDOWS.BT.zip)

I will try deleting softwaredistribution to see if that helps at all.

**Comment by AveYo** [comment]
@Core0101 your issue is different. That looks like `appraiserres.dll` has been deleted / replaced so the compat report is blank.
Are you doing that? Did you do that at some point? Could it be that you have the very old v1 script bypass installed?  
Get the v1 from the `bypass11\archive` folder and run it once. 
Does it say REMOVED or INSTALLED? 
Then regardless, run it again until it says REMOVED.

I used to auto-remove that in versions after it, but figured now nobody would be on that version anymore.
Guess I should get rid of the archive altogether, since older type of bypasses are no longer advised.

**Comment by Core0101** [comment]
Ah yup, I was using the old v1 script, I still had it on my PC so I ran it once and it said Removed. I actually ran into a big issue a couple days ago where explorer.exe would constantly crash and I couldn't get to anything. I tried deleting any updates that got installed, uninstalling my av, removing glasswire, sfc, dism, etc.. Until I thought maybe it was the tpm script interacting with the dev build or something so that's what brought me here. After installing v6 and rebooting my explorer worked again and stopped crashing. Then the next issue was just installing the updates. 

Anyways thank you for your help, I will check for updates again and report back.

**Comment by AveYo** [comment]
@Core0101 my instinct was right then ;)  
I have updated the script to force uninstall v1 wmi-based method as well _(already was uninstalling v2-v5 vdsldr-based)_  
And removed the archived folder with previous versions of the script

So now you just use the now v7 as is, install or remove it does not matter it will clear the previous ones.
Good luck!

**Comment by sam20112011** [comment]
Thank you man) You are doing an good and important thing. Thanks a lot again)

**Comment by samiran9200** [comment]
v7 Work like a charm, :-)

**Comment by Core0101** [comment]
![Screenshot 2021-12-10 202333](https://user-images.githubusercontent.com/95932132/145664793-923a120f-247c-4563-8fc6-964e783216fa.png)
Woo! It worked! Thanks so much again

**Comment by nondetect** [comment]
> @Core0101 my instinct was right then ;) I have updated the script to force uninstall v1 wmi-based method as well _(already was uninstalling v2-v5 vdsldr-based)_ And removed the archived folder with previous versions of the script
> 
> So now you just use the now v7 as is, install or remove it does not matter it will clear the previous ones. Good luck!

i told you about that 2 version of script early)

**Comment by AveYo** [comment]
@nondetect, 
You have access to a wider perspective from the russian community, where you probably seen the issue more commonly, 
so feel free to add new issues if you are aware of them, as I might overlook the importance of a comment in a commit.
At that time I could not believe that people would still be running v1.
I thought that people used v2, v3 having uninstall for v1 - since there were 2 new dev releases around that time as well.
**Bold** of me to assume people would bother checking script updates if it continued to work till 22509 - and it did.
After 22509, I did v6, but again not considered people might still be running v1 - that's my bad.
With proper feedback in this thread (logs) all is clear and finally addressed it with the hours-apart v6f - v7.
_And that's how microsoft has been bloating windows for the past decades - 4 lines to address x.y here, 40 there and so on_ 🥇 
_I now have to keep v1 and v2-v5 uninstall lines for as long as the script still works, despite unrelated new method used ;)_

**Comment by sam20112011** [comment]
Yes, that's right. I was on v.2 before update 22509 as everything worked as it should. The problem of the TPM 2.0 module is especially relevant in Russia, since their import into the country is regulated by the FSB. And due to the boom in mining, the shortage of semiconductors and the rise in prices for computer components, not everyone has the opportunity to afford processors with emulation of this module in BIOS.

**Comment by LillyWho** [comment]
> @LillyWho, are you sure you're running the v6? just run the script again, it should say:
> 
> > Skip TPM Check on Dynamic Update **v6** [INSTALLED] run again to remove
> 
> it might get blocked by a zealous antivirus - check your av logs
> 
> the whole code is added in the registry here: `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\SetupHost.exe` UseFilter = 0x1 `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\SetupHost.exe\0` Code= .. Debugger= .. FilterFullPath= .. make sure the entries are there

There was no AV other than Windows Defender, and I did have the latest bypass script. As it turns out it actually did upgrade, like I said, even though the setup window had said something failed, the download percentage did go up and the upgrade went through.

**Comment by AveYo** [comment]
@LillyWho, like I said a few times, previous method of bypasses worked by deleting `appraiserres.dll` or (in v5) file-deny `Appraiser_Data.ini`, and that was fine until 22509.   
Afterwards, windows update setup would fail, about 5-7 times, then it would reluctantly accept a blank compatibility report anyway and installation would complete.  
With v7 of the bypass, no files are deleted, the compatibility report is legit and windows update setup works from the first try.
So do run the v7 of the script, just to clear the previous versions you might have that will cause issues.

v7 should be more future-proof, but time will tell..

**Comment by LillyWho** [comment]
> @LillyWho, like I said a few times, previous method of bypasses worked by deleting `appraiserres.dll` or (in v5) file-deny `Appraiser_Data.ini`, and that was fine until 22509. Afterwards, windows update setup would fail, about 5-7 times, then it would reluctantly accept a blank compatibility report anyway and installation would complete. With v7 of the bypass, no files are deleted, the compatibility report is legit and windows update setup works from the first try. So do run the v7 of the script, just to clear the previous versions you might have that will cause issues.
> 
> v7 should be more future-proof, but time will tell..

It was _with the v7 script_ that's the joke.

**Comment by laracroftonline** [comment]
v7 worked fine on 22523

**Comment by owencz1998** [comment]
i have tried everything iv fresh install done the v7 but update says pending then does the restart but will not install is there anyone who could help that might off had the same issue :[

**Comment by AveYo** [comment]
@owencz1998 windows update can be unreliable on it's own - sometimes stuff breaks, specially with insider builds, and the only solution is to run an in-place repair from mounted iso media   

I can take a look at what happens in your case if you provide your `C:\$WINDOWS.~BT\Sources\Panther` logs (zip and attach / link)

**Comment by owencz1998** [comment]
[$WINDOWS.~BT.zip](https://github.com/AveYo/MediaCreationTool.bat/files/7766139/WINDOWS.BT.zip)
iv gone from a complete fresh install to repairs and constantly looking into the system but haven't got a clue 🤣🤣

**Comment by AveYo** [comment]
@owencz1998 looks like you're not running the latest v7 of the bypass?

**Comment by owencz1998** [comment]
oh I'm finding the v7 script that is highlighted green but not sure if I'm clicking on the correct one .

**Comment by AveYo** [comment]
that's the one.
then try again, it should work.
if not, provide the resulting `C:\$WINDOWS.~BT\Sources\Panther`

**Comment by owencz1998** [comment]
oh grate yes it works I didnt realise I clicked on the older v7 lol but all good what folder did u look at for the issues mate

**Comment by AveYo** [comment]
> That was showing both `/Product Server` _(bypass installed and active)_ and `/Product Client` _(bypass not working as intended)_, causing two compatibility reports with one obviously failing (the Client one) Why? because in v6 release I left a bad regex that did not replace launch options correctly. Funny thing, the installed bypass in my test machine had the updated regex so it did not fail for me. Sorry for the frustration caused by this.
> 
> @Core0101, @sam20112011 are you able to test the fixed version or have you updated already via iso ;)

for you it's the same as above. you had the short-lived v6 installed, did not update to v6f (fixed couple hours later) or the latest v7

**Comment by owencz1998** [comment]
getting new issue which is blue screen hypervisor error any reason why its started to do that 🤣

**Comment by AveYo** [comment]
that's unrelated, it happens for supported PCs as well  
usually the culprit is a conflict between virtualbox and hyper-v  
so uninstall virtualbox and/or disable hyper-v from Add/Remove Windows Features

**Comment by owencz1998** [comment]
iv not got virtualbox and hyper v is already disabled

**Comment by AveYo** [comment]
if you ever activated hyper-v / defender core isolation / **credential guard**, it's not as trivial to undo  

[uncheck those in optional features, disable via gpedit.msc, reboot and accept uefi prompt to turn it off](https://www.tenforums.com/tutorials/68935-enable-disable-credential-guard-windows-10-a.html)

**Comment by owencz1998** [comment]
ok thanks a lot :)
